app/components/polls/polls-edit/polls-edit-directive.js

```diff
@@ -0,0 +1,71 @@
+'use strict';
+
+angular.module('polls.edit.directive', [
+  'apiServices.poll.factory',
+  'BettyCropper',
+  'lodash',
+  'saveButton.directive',
+  'topBar'
+]).constant('RESPONSE_TYPES', [
+  {
+    name: 'Text Only',
+    value: 'Text'
+  },
+  {
+    name: 'Image + Text',
+    value: 'Image'
+  }
+])
+  .directive('pollsEdit', function (routes) {
+    return {
+      templateUrl: routes.COMPONENTS_URL + 'polls/polls-edit/polls-edit.html',
+      controller: function (_, $location, $q, $routeParams, $scope, $window, Poll) {
+        // populate model for use
+        if ($routeParams.id === 'new') {
+          $scope.model = Poll.$build();
+          $scope.isNew = true;
+        } else {
+          $scope.model = Poll.$find($routeParams.id);
+        }
+
+        var removeUnsavedChangesGuard = $window.addEventListener('onbeforeunload',  function (e) {
```

Comment:
can't call $on on $window, I could nest it though
`var removeUnsavedChangesGuard = function () {
    $window.onbeforeunload = function (e) { ...};
  };`